flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierUnaligner.java

```diff
@@ -113,7 +113,11 @@ public void processBarrier(CheckpointBarrier barrier, InputChannelInfo channelIn
 
 			if (++numBarriersReceived == numOpenChannels) {
 				allBarriersReceivedFuture.complete(null);
-				resetPendingCheckpoint(barrierId);
+				for (final InputGate gate : inputGates) {
+					for (int index = 0, numChannels = gate.getNumberOfInputChannels(); index < numChannels; index++) {
+						gate.getChannel(index).checkpointStopped(currentCheckpointId);
+					}
+				}
```

Comment:
What's the story behind this change? 
1. it seems it differs only by a single line `numBarriersReceived = 0;`, so at the very least we should deduplicate some code here
2. can you explain what's the functional change?
3. aren't we missing a unit test for that? It would help answer point 2., and if there was a bug discovered in e2e test, it would be nice to have a faster unit test for that as well.

Reply:
Sorry that was just a test commit to see if the stuck e2e failed because of this change. I removed it. The original change is covered by a few unit tests already.